From: Kevin Lo <kevlo@kevlo.org>
Date: Thu, 7 Jan 2021 14:10:38 +0800
Subject: igc: check return value of ret_val in igc_config_fc_after_link_up
Patch-mainline: v5.11-rc7
Git-commit: b881145642ce0bbe2be521e0882e72a5cebe93b8
References: bsc#1118657 FATE#325278

Check return value from ret_val to make error check actually work.

Fixes: 4eb8080143a9 ("igc: Add setup link functionality")
Signed-off-by: Kevin Lo <kevlo@kevlo.org>
Acked-by: Sasha Neftin <sasha.neftin@intel.com>
Signed-off-by: Tony Nguyen <anthony.l.nguyen@intel.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/intel/igc/igc_mac.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/intel/igc/igc_mac.c
+++ b/drivers/net/ethernet/intel/igc/igc_mac.c
@@ -667,7 +667,7 @@ s32 igc_config_fc_after_link_up(struct i
 	}
 
 out:
-	return 0;
+	return ret_val;
 }
 
 /**
